actors/market/tests/market_actor_test.rs

```diff
@@ -1501,6 +1503,45 @@ fn slash_a_deal_and_make_payment_for_another_deal_in_the_same_epoch() {
     check_state(&rt);
 }
 
+#[test]
+fn cron_reschedules_update_to_new_period() {
```

Comment:
I'm not 100% I understood, but added a test – please take a look. This strikes me as almost identical to the usual case of sequential re-schedulings of deals that are in the expected epochs.

Reply:
Yeah your test is what I had in mind.  In retrospect I agree it's pretty much the same.  If you'd prefer to remove it that's fine.  Also after this upgrade completes it might make sense to remove test cases around rescheduling misscheduled updates as this behavior will no longer be relevant.